src/tests/ftest/launch.py

```
@@ -897,6 +901,74 @@ def archive_logs(avocado_logs_dir, test_yaml, args):
+    """Get the size of the the directoy of the host test log files in
+    the avocado results directory and store the values in a file.
+
+    Args:
+        test_yaml (str): yaml file containing host names
+        test_file (str): the test python file
+        args (argparse.Namespace): command line arguments for this program
+        size_limit (int): limitsize in bytes that determines when to send
+            notification.
+    """
+    logs_dir = os.environ.get("DAOS_TEST_LOG_DIR", DEFAULT_DAOS_TEST_LOG_DIR)
+    log_size_file = os.path.join(logs_dir, "log_size.log")
+    host_list = get_hosts_from_yaml(test_yaml, args)
+
+    # Create a file that contains the sizes on each test log and the logs_dir.
+    command = "du -ab -d 1 {} &> {}".format(logs_dir, log_size_file)
+    spawn_commands(host_list, command, 30)
+
+    # Check log_size.log file to see if anything goes over size_limit
+    size_info_cmd = "cat {}".format(log_size_file)
+    size_info_out = get_remote_output(host_list, size_info_cmd, 30)
+
+    # Create a dictionary of hosts for each unique return code
+    results = {code: hosts for code, hosts in size_info_out.iter_retcodes()}
+
+    # Determine if the command completed successfully across all the hosts
+    status = len(results) == 1 and 0 in results
+    if not status:
+        print("  Errors detected running \"{}\":".format(command))
+
+    # Check the command output
+    for code in sorted(results):
+        output_data = list(size_info_out.iter_buffers(results[code]))
+        print("Out data: {}".format(output_data))
+        if not output_data:
+            err_nodes = NodeSet.fromlist(results[code])
+            print("    {}: rc={}, output: <NONE>".format(err_nodes, code))
+        else:
+            msg = []
+            for output, o_hosts in output_data:
+                for line in str(output).splitlines():
+                    if line.split("\t")[0] and line.split("\t")[0].isdigit():
```

I really appreciate that you are validating the contents of the file rather than just assuming they are valid, but if this `if` statement were to be false, then the log contents must be corrupt in some way, yes?

Would it be better to indicate this rather than just silently ignoring the line?

---

Not necessarily, I added a specific `if` because when I was testing I found that an initial "stty:" message would be appended at the beginning of the output like so:
```
stty: standard input: Inappropriate ioctl for device
7888    /var/tmp/daos_testing/daos.log
702     /var/tmp/daos_testing/daos_control.log
79      /var/tmp/daos_testing/log_size.log
1974    /var/tmp/daos_testing/daos_admin.log
14739   /var/tmp/daos_testing
```
But that line does not come from the file itself, it comes from the command to collect the contents of the file.

---

An `ssh $node cat $file` (which is all that clush does, in parallel) should not cause a `stty` error like that.  Can you reproduce it outside of Avocado?

---

I was able to reproduce locally and it might be something not configured correctly in my environment. I made a change to print out when the file has seemingly bad contents. Was debating on whether I should add more aggressive error handling if that were the case, my guess is that we wouldn't want to interrupt launch.py if log_size.log file is corrupted, so I just printed out a message.